File: flower/src/main/java/com/kharitonov/flower/entities/UserEntity.java
package com.kharitonov.flower.entities;

import com.kharitonov.flower.security.Role;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "usr")
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String login;
    private String password;
    private String firstName;
    private String lastName;
    @Enumerated(EnumType.STRING)
    private Role role;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn()
    private BasketEntity basket;



}

File: src/main/java/by/hghgrtut/shop/entities/UserEntity.kt
package by.hghgrtut.shop.entities

import by.hghgrtut.shop.security.Role
import javax.persistence.*

@Entity
@Table(name = "usr")
class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long? = null,
    val login: String? = null,
    val password: String? = null,
    val firstName: String? = null,
    val lastName: String? = null,
    @Enumerated(EnumType.STRING)
    val role: Role? = null,
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn
    val basket: BasketEntity? = null
}